### ISSUE TITLE
Fire and forget

### ISSUE THREAD

**Issue by andrei-popescu** [orange-cloudfoundry/cf-plugin-sync#4]
Right now the plugin functions as a file watcher. The problem is that the file watcher has to be "watched" so that it is always up. It's more often than not that the watcher is terminating resulting in files not being synchronised, creating frustration in development when one does not understand why his code doesn't work, when in fact its not being uploaded to the container.

Ideally this would be a "launch once, use always" type of a system, either launched by ```cf push``` itself, or launched after the ```cf push``` but without any process running from the source machine. 
Or any other version that does not require manual monitoring.

Ideally the mapping should be started and stopped, and automatically restarted when both the source (dev) and the target(container) become available.

Ex: 
```sh
#to start
cf sync -s . my-cf-app
#to stop
cf sync --end -s . my-cf-app
```

**Comment by ArthurHlt** [comment]
Thanks, I'm not sure about what you are saying by: 
> the watcher is terminating resulting in files not being synchronised

 is it linked to #2 ? 

About your proposal:
> either launched by `cf push` 

first o
In this way, this cannot be done. This actually means that a plugin creator could hijack official command to do his own things. If it was possible so many unfriendly behaviour could be created.

The other way would be polling app state to restart when needed watcher and synchronization. 

This is an interesting feature but it needs a lot of works for a "not that much" used plugin. I could reconsider to do it if more people will be interested by this feature.

Anyway, any PR are welcome to implements this functionality.

**Comment by andrei-popescu** [comment]
@ArthurHlt 
Yes, this is linked to #2. When the watcher stops, so does the file synchronisation. All files modified while the tool is off are not synchronised.

I think having a good way to develop is going to be a gain for everybody. Every single dev on the cloud is having the same issue and we're not few. We already have **tens of projects** in the cloud, and every single one is using this plugin to develop as the ```cf push``` is obviously way to slow. 

The main dev process should follow these steps:
1. Write code
2. Save code (the IDE does this automatically)
3. Test on container

Everything else should be as transparent as possible.
